pkg/eventmanager/eventmanager.go

```diff
@@ -104,111 +103,39 @@ func (s *eventManager) notificationRefresh() error {
 		}
 	}
 
-	// Check all the types of notifications we can send
-	err = checkUpgradeStart(s.metrics, s.notifier, uc)
-	if err != nil {
-		return err
-	}
-	err = checkUpgradeEnd(s.metrics, s.notifier, uc)
-	if err != nil {
-		return err
-	}
-	return nil
-}
-
-func checkUpgradeStart(mc metrics.Metrics, nc notifier.Notifier, uc *upgradev1alpha1.UpgradeConfig) error {
-
-	upgradeStarted := false
-
-	// Check if the cluster is at the version in the UpgradeConfig
-	isSet, err := mc.IsClusterVersionAtVersion(uc.Spec.Desired.Version)
-	if err != nil {
-		return fmt.Errorf("can't check cluster metric ClusterVersion: %v", err)
-	} else {
-		upgradeStarted = isSet
-	}
-
-	// As a backup if metrics is unavailable, check if upgradeConfig indicates the upgrade has completed
-	if !upgradeStarted {
-		upgradePhase, err := getUpgradePhase(uc)
-		if err == nil && *upgradePhase == upgradev1alpha1.UpgradePhaseUpgraded {
-			upgradeStarted = true
-		}
-	}
-
-	if !upgradeStarted {
-		return nil
-	}
-
-	// Check if a notification for it has already been sent successfully
-	isNotified, err := mc.IsMetricNotificationEventSentSet(uc.Name, string(notifier.StateStarted), uc.Spec.Desired.Version)
+	// Check if a notification for it has been sent successfully - if so, nothing to do
+	isNotified, err := s.metrics.IsMetricNotificationEventSentSet(uc.Name, string(state), uc.Spec.Desired.Version)
 	if err != nil {
 		return fmt.Errorf("can't check cluster metric NotificationSent: %v", err)
 	}
 	if isNotified {
 		return nil
 	}
 
-	// Send the notification and indicate it has been sent
-	description := fmt.Sprintf("Cluster is currently being upgraded to version %s", uc.Spec.Desired.Version)
-	err = nc.NotifyState(notifier.StateStarted, description)
-	if err != nil {
-		return fmt.Errorf("can't send notification '%s': %v", notifier.StateStarted, err)
+	// Customize the state description
+	var description string
+	switch state {
+	case notifier.StateStarted:
+		description = fmt.Sprintf("Cluster is currently being upgraded to version %s", uc.Spec.Desired.Version)
+	case notifier.StateDelayed:
+		description = fmt.Sprintf("Cluster is currently being upgraded to version %s", uc.Spec.Desired.Version)
```

Comment:
"Cluster upgrade to version %s is currently delayed".

Reply:
Whoops, thank you - have implemented this suggestion.